### ISSUE TITLE
Large StreamBuffer requests do not block until fulfilled

### ISSUE THREAD

**Issue by mbaz** [JuliaTelecom/SoapySDR.jl#17]
```
julia> x=read(stream,100000000);
┌ Info: assertion debugging
│   nread = 20072
└   n = 100000000
ERROR: AssertionError: nread == n
Stacktrace:
 [1] read(s::SoapySDR.Stream{ComplexF32}, n::Int64; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ SoapySDR ~/.julia/packages/SoapySDR/ElzKN/src/highlevel.jl:538
 [2] read(s::SoapySDR.Stream{ComplexF32}, n::Int64)
   @ SoapySDR ~/.julia/packages/SoapySDR/ElzKN/src/highlevel.jl:534
 [3] top-level scope
   @ REPL[40]:1
```
Would it make sense to take the requested number of samples as a maximum, and let `read` return less than that?

**Comment by sjkelly** [comment]
Yes I think that makes sense per `read` definition:
```
  read(s::IOStream, nb::Integer; all=true)

  Read at most nb bytes from s, returning a Vector{UInt8} of the bytes read.

  If all is true (the default), this function will block repeatedly trying to read all requested bytes, until an error or
  end-of-file occurs. If all is false, at most one read call is performed, and the amount of data returned is device-dependent.
  Note that not all stream types support the all option.
```

**Comment by sjkelly** [comment]
With #18 this can be temporarily avoided with `read(...; all=false)`, to return fewer than requested samples. We should handle this internally though, so let's leave this open to track.

**Comment by sjkelly** [comment]
To clarify, we should handle larger buffers internally in SoapySDR if requested, and block until the data become available. Just like the behavior of any other IO stream.

**Comment by sjkelly** [comment]
Should be fixed in #22. Please comment and reopen if otherwise.
